packages/SwingSet/src/kernel/kernel.js

```diff
@@ -1165,9 +1165,14 @@ export default function buildKernel(
       assert.fail(X`Kernel reentrancy is forbidden`);
     }
     processQueueRunning = Error('here');
-    return processor(message).finally(() => {
-      processQueueRunning = undefined;
-    });
+    try {
```

Comment:
I don't think having this `return ['crank-failed', []]` upon error will cause the caller (`step()`) to throw as we want.

How about:
```js
const p = processor(message);
p.finally(() => processQueueRunning = undefined);
p.catch(err => panic(...));
return p;
```
?

I think all the processors are async.

Reply:
The `return ['crank-failed', []]` is just there to make tsc/eslint shut up about the type of the return value.

The throw from `step()` et all is caused by having the `kernelPanic` flag be set by the call to `panic`.